lib/api/core/models/repositories/tokenRepository.js

```diff
@@ -8,94 +10,83 @@ var
   InternalError = require('kuzzle-common-objects').errors.InternalError,
   UnauthorizedError = require('kuzzle-common-objects').errors.UnauthorizedError;
 
-function parseTimespan(time) {
-  var milliseconds;
-
-  if (typeof time === 'string') {
-    milliseconds = ms(time);
-
-    if (typeof milliseconds === 'undefined') {
-      return -1;
-    }
-
-    return milliseconds;
-  }
-  else if (typeof time === 'number') {
-    return time;
-  }
-
-  return -1;
-}
-
 /**
  * @class TokenRepository
  * @extends Repository
- * @param {Kuzzle} kuzzle
- * @param opts
  * @constructor
  */
-function TokenRepository (kuzzle, opts) {
-  Repository.apply(this, arguments);
-  this.collection = 'token';
-  this.ObjectConstructor = Token;
-  if (opts !== undefined && opts.ttl !== undefined) {
-    this.ttl = opts.ttl;
+class TokenRepository extends Repository {
+  /**
+   * @param {Kuzzle} kuzzle
+   * @param opts
+   * @constructor
+   */
+  constructor (kuzzle, opts) {
+    super(kuzzle);
+    this.collection = 'token';
+    this.ObjectConstructor = Token;
+    if (opts !== undefined && opts.ttl !== undefined) {
+      this.ttl = opts.ttl;
+    }
   }
-}
 
-TokenRepository.prototype = Object.create(Repository.prototype);
-TokenRepository.prototype.constructor = TokenRepository;
-
-TokenRepository.prototype.init = function tokenRepositoryInit () {
-  Repository.prototype.init.call(this, {
-    databaseEngine: null
-  });
-};
-
-/**
- *
- * @param {Token} requestToken
- * @returns {Promise<Object>}
- */
-TokenRepository.prototype.expire = function tokenRepositoryExpire (requestToken) {
-  return Repository.prototype.expireFromCache.call(this, requestToken)
-    .then(() => this.kuzzle.tokenManager.expire(requestToken));
-};
-
-/**
- * @param {User} user
- * @param {Request} request
- * @param {object} opts
- * @returns {*}
- */
-TokenRepository.prototype.generateToken = function tokenRepositoryGenerateToken (user, request, opts) {
-  var
-    expiresIn,
-    token = new Token(),
-    options = opts || {},
-    encodedToken,
-    error;
-
-  if (!user || user._id === null) {
-    error = new InternalError('Unknown User : cannot generate token');
-    return Promise.reject(error);
+  init () {
+    super.init({});
```

Comment:
Token should be stored only into the Cache, not into the DatabaseEngine.
So, as `Repository` subclass is using the internalStorage as default database engine, we should explicitly disable it at init:
```javascript
init() {
  super.init({databaseEngine: null});
}
```